lib/nosedrum/interactor.ex

```diff
@@ -0,0 +1,143 @@
+defmodule Nosedrum.Interactor do
+  @moduledoc """
+  Interactors take the role of both `Nosedrum.Invoker` and `Nosedrum.Storage` when
+  it comes to Discord's Application Commands. An Interactor handles incoming
+  `t:Interaction.t/0`s, invoking `Nosedrum.ApplicationCommand.command/1` callbacks
+  and responding to the Interaction.
+
+  In addition to tracking commands locally for the bot, an Interactor is
+  responsible for registering an Application Command with Discord when `add_command/3`
+  or `remove_command/2` is called.
+  """
+
+  alias Nostrum.Snowflake
+  alias Nostrum.Struct.Interaction
```

Comment:
Adding an alias per my comment below:

```suggestion
  alias Nostrum.Struct.{Guild, Interaction}
```